rollup-plugins/export-in-place.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * Copyright 2019 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// This plugin is a workaround for a suboptimal format of Rollup
+// exports in ES module mode, where it produces code like:
+//
+// const A = 1, B = 2, ...;
+// export { A, B, ... };
+//
+// But for optimal size we can inline exports back as:
+//
+// export const A = 1, B = 2, ...;
+//
+// For now this plugin expects this specific form of exports
+// and isn't meant for general use, but still provides helpful size
+// optimisation for this project (~7% gzipped).
+
+import MagicString from "magic-string";
+import assert from "assert";
+
+export default function() {
+  return {
+    name: "export-in-place",
+
+    renderChunk(code) {
+      const ast = this.parse(code);
+      // Assert for the AST shape.
+      // 1) We should have only two declarations at top level.
+      assert.strictEqual(ast.body.length, 2);
```

Comment:
This means the plugin will throw when the AST has an unexpected shape, no?

I’d prefer to log a warning with `this.warn()` when that happens, as it’s still valid code.

Reply:
Sure; just wanted to make sure that it's easy to catch a case if output format ever changes.

Reply:
Added meaningful assert messages and a try-catch to just warn instead and fall back to the original source.